Several TypeInitializationExceptions with V2.10.4

[22:15:00] [ERROR] Caught fatal unhandled exception:
System.TypeInitializationException: Der Typeninitialisierer für "GTA.Native.MemoryAccess" hat eine Ausnahme verursacht. ---> System.NullReferenceException: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
   bei GTA.Native.MemoryAccess..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei GTA.Native.MemoryAccess.GetVehicleHandles()
   bei GTA.World.GetAllVehicles()
   bei JetBlast.ScriptMain.OnTick(Object sender, EventArgs e)
   bei GTA.Script.raise_Tick(Object value0, EventArgs value1)
   bei GTA.Script.MainLoop()
[22:15:00] [ERROR] Caught fatal unhandled exception:
System.TypeInitializationException: Der Typeninitialisierer für "GTA.Native.MemoryAccess" hat eine Ausnahme verursacht. ---> System.NullReferenceException: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
   bei GTA.Native.MemoryAccess..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei GTA.Native.MemoryAccess.GetPedHandles(Vector3 position, Single radius)
   bei GTA.World.GetNearbyPeds(Ped ped, Single radius)
   bei Main.OnTick(Object sender, EventArgs e)
   bei GTA.Script.raise_Tick(Object value0, EventArgs value1)
   bei GTA.Script.MainLoop()
[22:15:00] [ERROR] Caught fatal unhandled exception:
System.TypeInitializationException: Der Typeninitialisierer für "GTA.Native.MemoryAccess" hat eine Ausnahme verursacht. ---> System.NullReferenceException: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
   bei GTA.Native.MemoryAccess..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei GTA.Native.MemoryAccess.GetPropHandles(Vector3 position, Single radius)
   bei GTA.World.GetNearbyProps(Vector3 position, Single radius)
   bei SittingMod.Main.OnTick(Object sender, EventArgs e)
   bei GTA.Script.raise_Tick(Object value0, EventArgs value1)
   bei GTA.Script.MainLoop()
[22:15:00] [ERROR] Caught fatal unhandled exception:
System.TypeInitializationException: Der Typeninitialisierer für "GTA.Native.MemoryAccess" hat eine Ausnahme verursacht. ---> System.NullReferenceException: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
   bei GTA.Native.MemoryAccess..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei GTA.Native.MemoryAccess.GetPedHandles(Vector3 position, Single radius)
   bei GTA.World.GetNearbyPeds(Ped ped, Single radius)
   bei WorkingHotDogVendors.Main.OnTick(Object sender, EventArgs e)
   bei GTA.Script.raise_Tick(Object value0, EventArgs value1)
   bei GTA.Script.MainLoop()
[22:15:05] [ERROR] Script 'Main' is not responding! Aborting ...
[22:15:05] [DEBUG] Aborted script 'Main'.
[22:15:10] [ERROR] Script 'SittingMod.Main' is not responding! Aborting ...
[22:15:10] [DEBUG] Aborted script 'SittingMod.Main'.
[22:15:15] [ERROR] Script 'WorkingHotDogVendors.Main' is not responding! Aborting ...
[22:15:15] [DEBUG] Aborted script 'WorkingHotDogVendors.Main'.
[22:16:01] [ERROR] Caught fatal unhandled exception:
System.TypeInitializationException: Der Typeninitialisierer für "GTA.Native.MemoryAccess" hat eine Ausnahme verursacht. ---> System.NullReferenceException: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
   bei GTA.Native.MemoryAccess..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei GTA.Native.MemoryAccess.GetVehicleHandles(Int32 modelhash)
   bei GTA.World.GetAllVehicles(Model model)
   bei TheCharacterVehicleMod.VehicleTick(String characterFirstInitial, VehicleHash vehicleHash, Int32 blipColor, Vehicle firstPlayerCar, Vehicle secondPlayerCar)
   bei TheCharacterVehicleMod.OnTick(Object sender, EventArgs e)
   bei System.EventHandler.Invoke(Object sender, EventArgs e)
   bei GTA.Script.raise_Tick(Object value0, EventArgs value1)
   bei GTA.Script.MainLoop()
[22:16:06] [ERROR] Script 'TheCharacterVehicleMod' is not responding! Aborting ...
[22:16:06] [DEBUG] Aborted script 'TheCharacterVehicleMod'.
[22:16:07] [ERROR] Caught fatal unhandled exception:
System.TypeInitializationException: Der Typeninitialisierer für "GTA.Native.MemoryAccess" hat eine Ausnahme verursacht. ---> System.NullReferenceException: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
   bei GTA.Native.MemoryAccess..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei GTA.Native.MemoryAccess.IsModelAPed(Int32 modelHash)
   bei GTA.World.CreatePed(Model model, Vector3 position, Single heading)
   bei GTA.World.CreatePed(Model model, Vector3 position)
   bei Hangars.PropManager.InitMechanic()
   bei Hangars.HangarControl.OnTickEvent(Object sender, EventArgs e)
   bei GTA.Script.raise_Tick(Object value0, EventArgs value1)
   bei GTA.Script.MainLoop()
[22:16:12] [ERROR] Script 'Hangars.HangarControl' is not responding! Aborting ...
[22:16:12] [DEBUG] Aborted script 'Hangars.HangarControl'.

--- a/source/core/NativeMemory.cpp
+++ b/source/core/NativeMemory.cpp
@@ -154,10 +154,9 @@ namespace GTA
 			uintptr_t address;
 
 			// Get relative address and add it to the instruction address.
-			// 3 bytes equal the size of the opcode and its first argument. 7 bytes are the length of opcode and all its parameters.
-			address = FindPattern("\x33\xFF\xE8\x00\x00\x00\x00\x48\x85\xC0\x74\x58", "xxx????xxxxx");
-			GetAddressOfEntity = reinterpret_cast<uintptr_t(*)(int)>(*reinterpret_cast<int *>(address + 3) + address + 7);
-			address = FindPattern("\xB2\x01\xE8\x00\x00\x00\x00\x33\xC9\x48\x85\xC0\x74\x3B", "xxx????xxxxxxx");
+			address = FindPattern("\xE8\x00\x00\x00\x00\x48\x8B\xD8\x48\x85\xC0\x74\x2E\x48\x83\x3D", "x????xxxxxxxxxxx");
+			GetAddressOfEntity = reinterpret_cast<uintptr_t(*)(int)>(*reinterpret_cast<int *>(address + 1) + address + 5);
+			address = FindPattern("\xB2\x01\xE8\x00\x00\x00\x00\x48\x85\xC0\x74\x1C\x8A\x88", "xxx????xxxxxxx");
 			GetAddressOfPlayer = reinterpret_cast<uintptr_t(*)(int)>(*reinterpret_cast<int *>(address + 3) + address + 7);
 
 			address = FindPattern("\x48\xF7\xF9\x49\x8B\x48\x08\x48\x63\xD0\xC1\xE0\x08\x0F\xB6\x1C\x11\x03\xD8", "xxxxxxxxxxxxxxxxxxx");

